src/common/containers/build-details.js

```diff
@@ -43,33 +62,44 @@ class BuildDetails extends Component {
 
 }
 BuildDetails.propTypes = {
-  account: PropTypes.string.isRequired,
-  repo: PropTypes.string.isRequired,
-  fullName: PropTypes.string.isRequired,
+  repository: PropTypes.shape({
+    owner: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    fullName: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired
+  }),
   buildId: PropTypes.string.isRequired,
   build: PropTypes.object,
   isFetching: PropTypes.bool,
+  snapLink: PropTypes.string,
   error: PropTypes.object,
   dispatch: PropTypes.func.isRequired
 };
 
 const mapStateToProps = (state, ownProps) => {
-  const account = ownProps.params.account.toLowerCase();
-  const repo = ownProps.params.repo.toLowerCase();
+  const owner = ownProps.params.owner.toLowerCase();
+  const name = ownProps.params.name.toLowerCase();
+  const fullName = `${owner}/${name}`;
+  const url = getGitHubRepoUrl(fullName);
```

Comment:
Because we redirect to `/owner/name/builds` after SSO login or user can enter this URL directly, we don't have repository info in the store (from repository input), so it's constructed here based on `owner` and `name` from URL params.

I wonder if we could reuse functionality of reducer (that already does the same thing for repository input). But that would mean that before we can render builds details (or builds listing) we need to trigger an action to populate the store. Who should trigger this action? Component in `componentWillMount` ?